app/src/main/java/org/simple/clinic/medicalhistory/newentry/NewMedicalHistoryEffectHandler.kt

```diff
@@ -19,7 +31,42 @@ class NewMedicalHistoryEffectHandler @AssistedInject constructor(
   fun build(): ObservableTransformer<NewMedicalHistoryEffect, NewMedicalHistoryEvent> {
     return RxMobius
         .subtypeEffectHandler<NewMedicalHistoryEffect, NewMedicalHistoryEvent>()
-        .addConsumer(OpenPatientSummaryScreen::class.java, { effect -> uiActions.openPatientSummaryScreen(effect.patientUuid)}, schedulersProvider.ui())
+        .addConsumer(OpenPatientSummaryScreen::class.java, { effect -> uiActions.openPatientSummaryScreen(effect.patientUuid) }, schedulersProvider.ui())
+        .addTransformer(RegisterPatient::class.java, registerPatient(schedulersProvider.io()))
         .build()
   }
+
+  private fun registerPatient(scheduler: Scheduler): ObservableTransformer<RegisterPatient, NewMedicalHistoryEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(scheduler)
+          .flatMap { registerPatientEffect ->
+            val loggedInUser = userSession.loggedInUserImmediate()
+            requireNotNull(loggedInUser)
+
+            facilityRepository
+                .currentFacility(loggedInUser)
+                .take(1)
+                .map { facility ->
+                  val ongoingMedicalHistoryEntry = registerPatientEffect.ongoingMedicalHistoryEntry
+
+                  RegisterPatientData(loggedInUser, facility, ongoingMedicalHistoryEntry)
+                }
+          }
+          .flatMapSingle { (user, facility, ongoingMedicalHistoryEntry) ->
+            patientRepository.saveOngoingEntryAsPatient(user, facility)
+                .flatMap { registeredPatient ->
+                  medicalHistoryRepository
+                      .save(registeredPatient.uuid, ongoingMedicalHistoryEntry)
+                      .toSingleDefault(PatientRegistered(registeredPatient.uuid))
+                }
+          }
+    }
+  }
+
+  private data class RegisterPatientData(
```

Comment:
Can't this data class be a separate file? or will this be moved later?

Reply:
This was intentionally left in this class since it is not required anywhere else. The only reason this exists is so that we can group the required data together to pass to the next operator in the Rx chain.